### PR TITLE
Domain introspection run

### PR DESCRIPTION
**Pull request by lennyphan**
WLS Domain Introspection results (topology, situation config files, etc) are stored in a config map that is mounted on all managed server pods.  This change will:

1. Re-read topology from Config Map (if it exists) when Operator is restarted or event when the CRD is modified and we don't have the topology information cached.

2. Re-run the introspection when we have zero running managed server pods and we believe we will be starting one or more managed server pods.

### CODE REVIEW

File: operator/src/main/java/oracle/kubernetes/operator/DomainProcessorImpl.java [hunk 1]
```diff
@@ -893,17 +906,58 @@ public NextAction apply(Packet packet) {
...
+              + callResponse.getStatusCode());
+      return callResponse.getStatusCode() == CallBuilder.NOT_FOUND
+          ? onSuccess(packet, callResponse)
+          : super.onFailure(packet, callResponse);
+    }
+
+    @Override
+    public NextAction onSuccess(Packet packet, CallResponse<V1ConfigMap> callResponse) {
+      V1ConfigMap result = callResponse.getResult();
+      System.out.println("ReadSituConfigMapStep.onSuccess result: " + result);
```
Comment (alai8) [inline operator/src/main/java/oracle/kubernetes/operator/DomainProcessorImpl.java]: Should these System.out.println be replaced with LOGGER?

File: operator/src/main/java/oracle/kubernetes/operator/DomainProcessorImpl.java [hunk 2]
```diff
@@ -645,7 +649,8 @@ public StartPlanStep(DomainPresenceInfo info, Step next) {
 
     @Override
     public NextAction apply(Packet packet) {
-      Step strategy = new RegisterStep(info, getNext());
+      Step strategy =
+          Step.chain(readExistingSituConfigMap(info), new RegisterStep(info, getNext()));
```
Comment (rjeberhard) [inline operator/src/main/java/oracle/kubernetes/operator/DomainProcessorImpl.java]: RegisterStep still needs to be first.

File: operator/src/main/java/oracle/kubernetes/operator/DomainProcessorImpl.java [hunk 3]
```diff
@@ -893,17 +906,58 @@ public NextAction apply(Packet packet) {
...
+  private static class ReadSituConfigMapStep extends ResponseStep<V1ConfigMap> {
+    private final DomainPresenceInfo info;
+
+    ReadSituConfigMapStep(DomainPresenceInfo info) {
+      this.info = info;
+    }
+
+    @Override
+    public NextAction onFailure(Packet packet, CallResponse<V1ConfigMap> callResponse) {
+      System.out.println(
```
Comment (rjeberhard) [inline operator/src/main/java/oracle/kubernetes/operator/DomainProcessorImpl.java]: LOGGER.finer

File: operator/src/main/java/oracle/kubernetes/operator/DomainProcessorImpl.java (hunk 1 above)
Comment (rjeberhard) [reply operator/src/main/java/oracle/kubernetes/operator/DomainProcessorImpl.java]: Yes, LOGGER.finer

File: operator/src/main/java/oracle/kubernetes/operator/helpers/ConfigMapHelper.java [hunk 4]
```diff
@@ -266,12 +266,12 @@ public NextAction apply(Packet packet) {
       String result = (String) packet.remove(ProcessingConstants.DOMAIN_INTROSPECTOR_LOG_RESULT);
       // Parse results into separate data files
       Map<String, String> data = parseIntrospectorResult(result, domain.getDomainUID());
-      LOGGER.fine("================");
-      LOGGER.fine(data.toString());
-      LOGGER.fine("================");
+      System.out.println("================");
```
Comment (rjeberhard) [inline operator/src/main/java/oracle/kubernetes/operator/helpers/ConfigMapHelper.java]: Put back :)

File: operator/src/main/java/oracle/kubernetes/operator/helpers/JobHelper.java [hunk 5]
```diff
@@ -100,12 +107,66 @@ public DomainIntrospectorJobStep(Step next) {
...
-      packet.putIfAbsent(START_TIME, Long.valueOf(System.currentTimeMillis()));
+      return doNext(getNext(), packet);
+    }
+  }
 
-      return doNext(context.createNewJob(getNext()), packet);
+  private static boolean runIntrospector(DomainPresenceInfo info) {
+    Domain dom = info.getDomain();
+    Scan scan = ScanCache.INSTANCE.lookupScan(dom.getMetadata().getNamespace(), dom.getDomainUID());
+    System.out.println("++++++ JobHelper.runIntrospector scan: " + scan);
```
Comment (rjeberhard) [inline operator/src/main/java/oracle/kubernetes/operator/helpers/JobHelper.java]: More...

File: operator/src/main/java/oracle/kubernetes/operator/DomainProcessorImpl.java (hunk 1 above)
Comment (lennyphan) [reply operator/src/main/java/oracle/kubernetes/operator/DomainProcessorImpl.java]: Removed or replaced the System.out.println debug statements. Just pushed the changes

**Pull request merged**